pepper-apis/src/main/resources/org/broadinstitute/ddp/db/dao/ActivityAnswerCopierSql.sql.stg

```diff
@@ -0,0 +1,101 @@
+group ActivityAnswerCopierSql;
+
+getLatestAnswer() ::= <<
+select
+src_a.answer_id,
+src_qt.question_type_code
+from
+answer src_a
+join question as src_q on src_q.question_id = src_a.question_id
+join question_type src_qt on src_qt.question_type_id = src_q.question_type_id
+join activity_instance as src_ai on src_ai.activity_instance_id = src_a.activity_instance_id
+join study_activity as src_act on src_act.study_activity_id = src_ai.study_activity_id
+join question_stable_code as src_q_code on src_q_code.question_stable_code_id = src_q.question_stable_code_id
+where
+src_ai.activity_instance_id = :activityInstanceId
+and
+src_q_code.stable_id = :questionStableId
+order by src_a.last_updated_at desc limit 1
+>>
+
+copyDateAnswer() ::= <<
+insert into date_answer(answer_id, day, month, year)
+(select :destinationAnswerId,
+a.day,
+a.month,
+a.year
+from
+date_answer a
+where
+a.answer_id = :sourceAnswerId)
+>>
+
+copyTextAnswer() ::= <<
+insert into text_answer(answer_id, answer)
+(select :destinationAnswerId,
+a.answer
+from
+text_answer a
+where
+a.answer_id = :sourceAnswerId)
+>>
+
+copyAgreementAnswer() ::= <<
+insert into agreement_answer(answer_id, answer)
+(select :destinationAnswerId,
+a.answer
+from
+agreement_answer a
+where
+a.answer_id = :sourceAnswerId)
+>>
+
+copyBooleanAnswer() ::= <<
+insert into boolean_answer(answer_id, answer)
+(select :destinationAnswerId,
+a.answer
+from
+boolean_answer a
+where
+a.answer_id = :sourceAnswerId)
+>>
+
+copyNumericAnswer() ::= <<
+insert into numeric_answer(answer_id, int_value)
+(select :destinationAnswerId,
+a.int_value
+from
+numeric_answer a
+where
+a.answer_id = :sourceAnswerId)
+>>
+
+
+copyPicklistAnswer() ::= <<
+insert into picklist_option__answer(answer_id, picklist_option_id, detail_text)
+(select :destinationAnswerId,
+a.picklist_option_id,
+a.detail_text
+from
+picklist_option__answer a
+where
+a.answer_id = :sourceAnswerId)
+>>
+
+baseActivityAnswerCopy() ::= <<
+insert into answer(answer_guid,question_id,operator_user_id,activity_instance_id,created_at,last_updated_at)
+(select
+:answerGuid,
+dest_q.question_id,
+src_a.operator_user_id,
+dest_ai.activity_instance_id,
+:createdAt,
+:lastUpdatedAt
+from
+answer src_a
+join question_stable_code as dest_q_code on dest_q_code.stable_id = :destinationStableId
+join question as dest_q on dest_q.question_stable_code_id = dest_q_code.question_stable_code_id
+join activity_instance as dest_ai on dest_ai.activity_instance_id = :destinationActivityInstanceId and dest_q.study_activity_id = dest_ai.study_activity_id
```

Comment:
I'm a bit worried about this piece of sql. Question stableId is not unique enough -- we'll need to restrict by study id as well. We can probably get that by joining over some tables using `sourceInstanceId`.

Also, probably not an issue now but what if there's different revisions of the question? We probably need to restrict it down to the question_id that matches up with the `destinationInstanceId`.

Reply:
Nice catch--I'll add a few more criteria and for both uniqueness and revisioning.